sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -146,7 +146,10 @@ trait CodegenSupport extends SparkPlan {
       if (outputVars != null) {
         assert(outputVars.length == output.length)
         // outputVars will be used to generate the code for UnsafeRow, so we should copy them
-        outputVars.map(_.copy())
+        outputVars.map(_.copy()) match {
```

Comment:
What in `consume` is relying on a side effect of traversing the `outputVars`?

Reply:
What I found is that in `prepareRowVar` the `evaluateVariables` call clears the code of `outputVars`.